test/WebJobs.Script.Tests.Integration/Storage/AzureStorageProviderTests.cs

```diff
@@ -0,0 +1,234 @@
+            {
+                { "ConnectionStrings:AzureWebJobsStorage", Environment.GetEnvironmentVariable(StorageConnection) },
+                { "AzureWebJobsStorage", "" }
+            };
+
+            using (new TestScopedEnvironmentVariable(testData))
+            {
+                var webHostConfiguration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddTestSettings()
+                    .Build();
+
+                var jobHostConfiguration = new ConfigurationBuilder().Build();
+
+                var azureStorageProvider = GetAzureStorageProvider(webHostConfiguration, jobHostConfiguration);
+                var container = azureStorageProvider.GetBlobContainerClient();
+                await VerifyContainerClientAvailable(container);
+
+                Assert.True(azureStorageProvider.TryGetBlobServiceClientFromConnection(out BlobServiceClient blobServiceClient, ConnectionStringNames.Storage));
+                await VerifyBlobServiceClientAvailable(blobServiceClient);
+            }
+        }
+
+        [Fact]
+        public async Task TestAzureStorageProvider_ConnectionInJobHostConfiguration()
+        {
+            var testData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "ConnectionStrings:AzureWebJobsStorage", Environment.GetEnvironmentVariable(StorageConnection) },
+                { "AzureWebJobsStorage", "" }
+            };
+
+            using (new TestScopedEnvironmentVariable(testData))
+            {
+                var webHostConfiguration = new ConfigurationBuilder().Build();
+
+                var jobHostConfiguration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddTestSettings()
+                    .Build();
+
+                var azureStorageProvider = GetAzureStorageProvider(webHostConfiguration, jobHostConfiguration);
+                var container = azureStorageProvider.GetBlobContainerClient();
+                await VerifyContainerClientAvailable(container);
+
+                Assert.True(azureStorageProvider.TryGetBlobServiceClientFromConnection(out BlobServiceClient blobServiceClient, ConnectionStringNames.Storage));
+                await VerifyBlobServiceClientAvailable(blobServiceClient);
+            }
+        }
+
+        [Fact]
+        public void TestAzureStorageProvider_NoConnectionThrowsException()
+        {
+            var testData = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "AzureWebJobsStorage", "" }
+            };
+
+            using (new TestScopedEnvironmentVariable(testData))
+            {
+                var webHostConfiguration = new ConfigurationBuilder()
+                    .AddEnvironmentVariables()
+                    .AddTestSettings()
+                    .Build();
```

Comment:
Nit -- You should be able to replace these lines (here and elsewhere) with `TestHelpers.GetTestConfiguration()`.